src/image-generation-service.ts

```diff
@@ -63,6 +63,17 @@ export class ImageGenerationService {
       if (imgCount === 0) {
         continue;
       }
+      const regex = new RegExp(CONFIG['Ad Group Name Regex']);
+      const matchGroups = this.getRegexMatchGroups(adGroup.adGroup.name, regex);
+      let prompt;
+      if (matchGroups) {
+        prompt = this.createPrompt(matchGroups);
+      } else {
+        Logger.log(
+          `No matching groups found for ${adGroup.adGroup.name} with ${regex}. Using full prompt.`);
+        prompt = CONFIG['ImgGen Prompt'];
+      }
+      const images = this._vertexAiApi.callVisionApi(adGroup.adGroup.name, prompt, imgCount);
```

Comment:
It looks like the next line of code overwrites the value of the `images` variable. I guess you don't need the next line.